src/main.py

```diff
@@ -0,0 +1,53 @@
+import io
+import streamlit as st
+
+from PIL import Image
+from img_to_text_gen import IttGenerator
+
+
+def gen_text(image):
+
+    itt_generator = IttGenerator()
+
+    text = itt_generator.gen(image=image)
+
+    if text is None:
+        return 'Failed wile generating text from image'
+
+    return text
+
+
+def main():
+    st.set_page_config(
+        page_title='Funny descriptions generator',
+        page_icon=':skull:',
+        layout='wide')
+
+    itt_generator = IttGenerator()
+    itt_generator.num_beams = st.sidebar.slider("Num beams", 1, 16, 4)
+    itt_generator.max_length = st.sidebar.slider("Max length", 64, 512, 128)
+    itt_generator.min_length = st.sidebar.slider("Min length", 1, 64, 16)

```

Comment:
This is not the best interaction between objects. Someone needs separately set generation values and after that call the generation method. There would be problems with checking the invariance since you don't have proper setters and with adding new interfaces for your app.

I suggest passing these values in the `gen` method along with an image.